movies/src/components/filter/ResultFilter.js

```diff
@@ -1,53 +1,57 @@
-import React, { useState } from 'react';
+import React, {useCallback, useState} from 'react';
 import PropTypes from 'prop-types';
+import {connect} from "react-redux";
+import {bindActionCreators} from "redux";
 import Constants from '../constants';
+import {changeFiltering} from "../../store/actions/actions";
 
-export default function ResultFilter(props) {
-  const genres = Constants.GENRE;
-  const { updateMovies } = props;
-  const [activeGenre, setActiveGenre] = useState(genres[0]);
-
-  const sortMovieList = (movieGenre) => {
-    const movieList = Constants.MOVIES;
-    if (movieGenre.id === genres[0].id) {
-      updateMovies(movieList);
-      setActiveGenre(genres[0]);
-    } else {
-      const currentGenre = movieGenre.title;
+const mapStateToProps =(state) => {
+  return {
+    movies: state,
+    filteringType: state
+  }
+};
 
-      const sortedMovieList = movieList.filter((movie) => movie.genre.some(
-        (genre) => genre.toLowerCase === activeGenre,
-      ));
+const matchDispatchToProps = (dispatch) => {
+  return bindActionCreators({updateMovies: changeFiltering}, dispatch)
+};
 
-      const activeGenreIndex = genres.findIndex((el) => el.title === currentGenre);
-      setActiveGenre(genres[activeGenreIndex]);
 
-      updateMovies(sortedMovieList);
-    }
-  };
+const ResultFilter = (props) => {
+  const genres = Constants.GENRE;
+  const { updateMovies } = props;
+  const [activeGenre] = useState(genres[0]);
 
-  const filterMovies = (genre) => {
-    sortMovieList(genre);
-  };
+  const filterMovies = useCallback((e) => {
+    const genre = e.target.dataset && e.target.dataset.name;
+    Array.from(document.querySelectorAll('.genre-filter__item')).forEach(item => {
+      item.classList.remove('active');
```

Comment:
avoid direct DOM manipulations when you're using react. The only source of dom change should be react state or props